src/ucp/core/ucp_worker.inl

```diff
@@ -258,41 +268,57 @@ ucp_worker_get_rkey_config(ucp_worker_h worker, const ucp_rkey_config_key_t *key
     return ucp_worker_add_rkey_config(worker, key, cfg_index_p);
 }
 
-#define UCP_WORKER_GET_EP_BY_ID(_worker, _ep_id, _action, _fmt_str, ...) \
-    ({ \
-         ucp_ep_h __ep = ucp_worker_get_ep_by_id(_worker, _ep_id); \
-         if (ucs_unlikely(__ep == NULL)) { \
-             ucs_trace_data("worker %p: ep id 0x%" PRIx64 " was not found, drop" \
-                            _fmt_str, _worker, _ep_id, ##__VA_ARGS__); \
-             _action; \
-         } \
-         __ep; \
-    })
-
-#define UCP_WORKER_GET_VALID_EP_BY_ID(_worker, _ep_id, _action, _fmt_str, ...) \
-    ({ \
-         ucp_ep_h ___ep = UCP_WORKER_GET_EP_BY_ID(_worker, _ep_id, _action, \
-                                                  _fmt_str, ##__VA_ARGS__); \
-         if (ucs_unlikely((___ep != NULL) && \
-                          (___ep->flags & UCP_EP_FLAG_CLOSED))) { \
-             ucs_trace_data("worker %p: ep id 0x%" PRIx64 " was already closed" \
-                            " ep %p, drop " _fmt_str, _worker, _ep_id, ___ep, \
-                            ##__VA_ARGS__); \
-             _action; \
-         } \
-         ___ep; \
-    })
-
-#define UCP_WORKER_GET_REQ_BY_ID(_worker, _req_id, _action, _fmt_str, ...) \
-    ({ \
-         ucp_request_t *_req = ucp_worker_get_request_by_id(_worker, _req_id); \
-         if (ucs_unlikely(_req == NULL)) { \
-             ucs_trace_data("worker %p: req id 0x%" PRIx64 " doesn't exist" \
-                            " drop " _fmt_str, _worker, _req_id, \
-                            ##__VA_ARGS__); \
-             _action; \
-         } \
-         _req; \
-    })
+#define UCP_WORKER_GET_EP_BY_ID(_ep_p, _worker, _ep_id, _action, _fmt_str, ...) \
+    { \
+        ucs_status_t __status; \
+        \
+        __status = ucp_worker_get_ep_by_id(_worker, _ep_id, _ep_p); \
+        if (ucs_unlikely(__status != UCS_OK)) { \
+            ucs_trace_data("worker %p: ep id 0x%" PRIx64 \
+                           " was not found, drop" _fmt_str, \
+                           _worker, _ep_id, ##__VA_ARGS__); \
+            _action; \
+        } \
+    }
+
+#define UCP_WORKER_GET_VALID_EP_BY_ID(_ep_p, _worker, _ep_id, _action, \
+                                      _fmt_str, ...) \
+    { \
+        UCP_WORKER_GET_EP_BY_ID(_ep_p, _worker, _ep_id, _action, _fmt_str, \
+                                ##__VA_ARGS__); \
+        if (ucs_unlikely((*(_ep_p))->flags & UCP_EP_FLAG_CLOSED)) { \
```

Comment:
excessive `()` around the condition?

Reply:
IMO it's needed since there is also the () of ucs_unlikely

Reply:
I see, yes